odd_collector_azure/adapters/azure_data_factory/mapper/activity.py

```diff
@@ -0,0 +1,70 @@
+from collections import deque
+
+from funcy import lpluck_attr
+from odd_collector_sdk.utils.metadata import DefinitionType, extract_metadata
+from odd_models import DataEntity, DataEntityGroup, DataEntityType, DataTransformer
+from oddrn_generator import AzureDataFactoryGenerator
+
+from odd_collector_azure.adapters.azure_data_factory.domain import ADFActivity
+
+from ..utils import ADFMetadataEncoder
+
+
+def map_activity(
+    oddrn_generator: AzureDataFactoryGenerator, activity: ADFActivity
+) -> deque[DataEntity]:
+    entities = deque()
+    activities = deque()
+
+    if hasattr(activity, "activities") and activity.activities:
+        for act in activity.activities:
+            activities.extend(
+                map_activity(
+                    oddrn_generator,
+                    ADFActivity(act, all_activities=activity.activities),
+                )
+            )
+
+    inputs = (
+        [
+            oddrn_generator.get_oddrn_by_path("activities", act)
+            for act in activity.inputs
+        ]
+        if hasattr(activity, "inputs") and activity.inputs
```

Comment:
Why do you need to check `hasattr` if ADFActivity has that field as a property and by contract returns list?